node/impl/client/client.go

```diff
@@ -289,6 +292,7 @@ func (a *API) newDealInfoWithTransfer(transferCh *api.DataTransferChannel, v sto
 		Verified:          v.Proposal.VerifiedDeal,
 		TransferChannelID: v.TransferChannelID,
 		DataTransfer:      transferCh,
+		DealStages:        v.DealStages,
```

Comment:
This data can be quite heavy to transfer over the wire (JSON-RPC) if it's not needed.

Since the CLI only allows to inspect a single deal, why not create an `ClientDealInspect` JSON-RPC operation? It's wasteful to send over all the DealStages detail, when we know the CLI will only use a single one.

Reply:
How big can this get?

Reply:
If the deal transfer stages are unbounded (i.e. data transfer keeps crashing, and deal is restarted indefinitely), this will be unbounded. In general for successful deals (or those that have a few failures), the struct looks like the following (see attached file).
[output.txt](https://github.com/filecoin-project/lotus/files/6243450/output.txt)

Reply:
So this is ok for a single deal, but I think that we shouldn't set this in `ClientListDeals` (which we do now)

Reply:
I am now populating the Stages and DealStages fields only when calling `ClientGetDealInfo` and not on `ClientListDeals`.